packages/zendesk-support-adapter/src/filters/user.ts

```diff
@@ -0,0 +1,236 @@
+    ...[
+      { fieldName: 'conditions.all', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }, { name: 'role' }] },
+      { fieldName: 'conditions.any', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }, { name: 'role' }] },
+    ],
+  ]),
+  view: mergeUserReplacers([
+    replaceConditionsAndActionsCreator(DEFAULT_REPLACER_PARAMS_FOR_CONDITIONS),
+    replaceRestriction,
+  ]),
+  workspace: workspaceReplacer,
+}
+
+const getUsers = async (paginator: clientUtils.Paginator): Promise<User[]> => {
+  const paginationArgs = {
+    url: '/users?role[]=admin&role[]=agent',
+    paginationField: 'next_page',
+  }
+  const users = (await toArrayAsync(
+    paginator(paginationArgs, page => makeArray(page) as clientUtils.ResponseValue[])
+  )).flat().flatMap(response => response.users)
+  if (!areUsers(users)) {
+    return []
+  }
+  return users
+}
+
+const thereAreRelevantChangesToReplace = (changes: Change<InstanceElement>[]): boolean => {
+  const changesTypeNames = _.uniq(changes.map(getChangeData).map(e => e.elemID.typeName))
+  return !_.isEmpty(_.intersection(changesTypeNames, Object.keys(TYPE_NAME_TO_REPLACER)))
+}
+
+/**
+ * Replaces the user ids with emails
+ */
+const filterCreator: FilterCreator = ({ paginator }) => {
+  let userIdToEmail: Record<string, string> = {}
+  let emailToUserId: Record<string, string> = {}
+  const deployModificationFunc = async (
+    changes: Change<InstanceElement>[],
+    mapping: Record<string, string>,
+  ): Promise<void> => {
+    const changesTypeNames = _.uniq(changes.map(getChangeData).map(e => e.elemID.typeName))
+    if (_.isEmpty(_.intersection(changesTypeNames, Object.keys(TYPE_NAME_TO_REPLACER)))) {
+      return
+    }
+    changes.forEach(change => {
+      _.mapValues(change.data, (instance: InstanceElement) => {
+        TYPE_NAME_TO_REPLACER[instance.elemID.typeName]?.(instance, mapping)
+      })
+    })
+  }
+  return {
+    onFetch: async elements => {
+      const users = await getUsers(paginator)
+      if (_.isEmpty(users)) {
+        return
+      }
+      userIdToEmail = Object.fromEntries(
+        users
+          .map(user => [user.id.toString(), user.email])
+      ) as Record<string, string>
+      const instances = elements.filter(isInstanceElement)
+      instances.forEach(instance => {
+        TYPE_NAME_TO_REPLACER[instance.elemID.typeName]?.(instance, userIdToEmail)
+      })
+    },
+    preDeploy: async (changes: Change<InstanceElement>[]) => {
+      if (!thereAreRelevantChangesToReplace(changes)) {
+        return
+      }
+      const users = await getUsers(paginator)
+      if (_.isEmpty(users)) {
+        return
+      }
+      userIdToEmail = Object.fromEntries(
+        users.map(user => [user.id.toString(), user.email])
+      ) as Record<string, string>
+      emailToUserId = Object.fromEntries(
```

Comment:
why is this stored as a context variable?
it is never used outside this callback
```suggestion
      const emailToUserId = Object.fromEntries(
```